bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -225,4 +258,25 @@ private Sitemap getSitemap(String sitemapName) {
         return null;
     }
 
+    @Override
+    public void modelChanged(String modelName, EventType type) {
+        if (type != EventType.MODIFIED || !modelName.endsWith(SITEMAP_SUFFIX)) {
+            return; // we process only sitemap modifications here
+        }
+
+        String changedSitemapName = StringUtils.removeEnd(modelName, SITEMAP_SUFFIX);
+
+        for (Entry<String, PageChangeListener> listenerEntry : pageChangeListeners.entrySet()) {
+            String sitemapWithPage = listenerEntry.getKey();
+            String sitemapName = extractSitemapName(sitemapWithPage);
+            String pageId = extractPageId(sitemapWithPage);
+
+            if (sitemapName.equals(changedSitemapName)) {
+                EList<Widget> widgets = collectWidgets(sitemapName, pageId);
```

Comment:
inline `pageId`-> `extractPageId(sitemapWithPage)`
Not that it would give us super-light-speed, but anyway, it's only needed here